source/extensions/filters/network/redis_proxy/proxy_filter.cc

```diff
@@ -192,11 +192,13 @@ void ProxyFilter::onAsyncResponse(Common::Redis::RespValuePtr&& value){
   }
 
   // Check if there is an active transaction that needs to be closed.
-  if (transaction_.should_close_ ) {
+  if (transaction_.should_close_ || transaction_.is_blocking_command_) {
       //Close all upsteam clients and ref to pubsub callbacks if any
       transaction_.close();
       // decrement the reference to proxy filter
-      transaction_.setPubsubCallback(nullptr);
+      if (!transaction_.is_blocking_command_ ) {
+        transaction_.setPubsubCallback(nullptr);
```

Comment:
This has to set irrespective of any command type , i wil fix it as part of my fixes

Reply:
Okay